time pubTime
int32 arraySize
float64[] effort
